hadoop-plugin/src/main/groovy/com/linkedin/gradle/hdfs/HdfsFileSystem.groovy

```diff
@@ -250,4 +295,37 @@ class HdfsFileSystem {
     logger.info("mkdir called on path ${p.toString()}")
     return fs.mkdirs(p);
   }
-}
\ No newline at end of file
+
+  /**
+   * Utility function to copy zipIn inputstream to file specified by filePath.
+   *
+   * @param zipIn Inputstream from a zip file
+   * @param filePath File path where zipIn should be written
+   * @throws IOException If zipIn cannot be written to filePath
+   */
+  private extractFile(ZipInputStream zipIn, String filePath) throws IOException {
+
+    // get output stream from the filesystem
+    OutputStream out = fs.create(new Path(filePath));
+
+    InputStream is = zipIn;
+
+    // get the name of the file
+    String fileName = filePath.substring(filePath.lastIndexOf("/")+1);
+
+    logger.info("copying ${fileName} to ${filePath}");
+
+    try {
+      // copy the file to the hdfs
+      IOUtils.copyBytes(is, out, conf, false);
```

Comment:
I've verified kerberos authentication. It works.

It uses webhdfs since we are using the webhdfs object(fs) that we have created while initializing this HdfsFileSystem. I think this is also the reason why Kerberos authentication works.